nixos/modules/tasks/auto-upgrade.nix

```diff
@@ -82,11 +98,23 @@ let cfg = config.system.autoUpgrade; in
 
   config = lib.mkIf cfg.enable {
 
-    system.autoUpgrade.flags =
-      [ "--no-build-output" ]
-      ++ (if cfg.channel == null
-          then [ "--upgrade" ]
-          else [ "-I" "nixpkgs=${cfg.channel}/nixexprs.tar.xz" ]);
+    assertions = [{
+      assertion = !((cfg.channel != null) && (cfg.flake != null));
+      message = ''
+        Updating nixos with the use of flakes cannot use channels.
+      '';
+    }];
+
+    system.autoUpgrade.flags = [ "--no-build-output" ]
+      ++ (if cfg.flake == null then
+        (if cfg.channel == null then
+          [ "--upgrade" ]
+        else [
+          "-I"
+          "nixpkgs=${cfg.channel}/nixexprs.tar.xz"
+        ])
+      else
+        [ "--flake ${cfg.flake}" ]);
```

Comment:
Does this also updates flakes automatically? I thought one has to use `nix flake update --update-input nixos-hardware`?

Reply:
No, just updates from the flake Uri. So the lock file from the flake Uri is used as in `nixos-rebuild --flake <uri> switch`

Reply:
Is this actually useful? I thought the intent of this module is to install the latest nixpkgs channel updates.

Reply:
For me it's useful. I update my nixfiles repo, and a day later every server in my network has the version as defined in the nixfiles repo lockfile

Reply:
If you're using a local repo, you can probably set `system.autoUpgrade.flags = [ "--update-input" "nixpkgs" "--commit-lock-file" ]` to update Nixpkgs.

Reply:
Ok. Would it not make sense to make this the default or at least add it to examples?

Reply:
added it to the description to not break systems without flake support.